dominion/src/main/java/org/codice/dominion/options/karaf/KarafOptions.java

```diff
@@ -0,0 +1,288 @@
+     * Per default the framework simply copies all referenced artifacts (via {@link
+     * org.codice.dominion.options.Options} and {@link KarafOptions} to the deploy folder of the
+     * karaf (base) distribution. If you don't have such a folder (for any reason) you can set this
+     * option to <code>false</code>. The Dominion container framework will then try to add those
+     * deployment urls directly to a features xml instead of copying those files to the deploy
+     * folder.
+     *
+     * @return <code>false</code> to have the Dominion container framework add deployment urls
+     *     directly to a features xml file; "org.apache.karaf.main.Main" to have it copy those
+     *     deployment files to the deploy folder
+     */
+    boolean deploy() default false;
+
+    /**
+     * Per default the Dominion container framework will run Karaf as a forked Java process. This
+     * can be used to run Karaf as an embedded instance instead.
+     *
+     * @return <code>false</code> to run Karaf as a forked Java process; <code>true</code> to run it
+     *     as an embedded instance
+     */
+    boolean embed() default false;
+
+    /**
+     * Specifies the Karaf platform these options are for (defaults to the platform where PaxExam is
+     * running). These options will be ignored if the Dominion container framework is not running on
+     * the same platform.
+     *
+     * @return the platform these options are for
+     */
+    Platform platform() default Platform.DEFAULT;
+
+    /**
+     * Specifies the startup script relative to <code>{karaf.home}</code> for the framework
+     * (defaults to either <code>bin\\Karaf.bat</code> or <code>bin/karaf</code> based on the
+     * current OS).
+     *
+     * <p><i>Note:</i> This script will automatically be made executable.
+     *
+     * @return the relative startup script filename for the framework
+     */
+    @Interpolate
+    String start() default Options.NOT_DEFINED;
+
+    /**
+     * Specifies file(s) relative to <code>{karaf.home}</code> to make executable before starting up
+     * the framework.
+     *
+     * <p><i>Note:</i> The startup script and all files located under <code>{karaf.bin}</code> are
+     * automatically made executable so there is no need to specify them again here.
+     *
+     * @return relative filename(s) to make executable
+     */
+    @Interpolate
+    String[] executables() default {};
+
+    /** Type of platforms the Karaf framework can run in. */
+    public enum Platform {
+      WINDOWS,
+      UNIX,
+      DEFAULT
+    }
+  }
+
+  /**
+   * Option that ensures that information about the overridden location of the local Maven
+   * repository user using the <code>maven.repo.local</code> will be passed to the container.
+   */
+  @Conditions.NotBlankSystemProperty("maven.repo.local")
+  @UpdateConfigFile(
+    target = KarafOptions.PAX_URL_MVN_CFG,
+    key = "org.ops4j.pax.url.mvn.localRepository",
+    value = "{maven.repo.local:-}"
```

Comment:
❓ Why does this have a `:-` on the end of it?

Reply:
✏️ Document `:-` in Interpolate.

Reply:
It adds support for defaulting an expression to something when the referenced system property is not defined or is null. 

I updated the documentation.